crates/ide-db/src/apply_change.rs

```diff
@@ -1,19 +1,14 @@
 //! Applies changes to the IDE state transactionally.
 
 use profile::Bytes;
-use salsa::{Database as _, Durability};
+use salsa::Database as _;
 
 use crate::{ChangeWithProcMacros, RootDatabase};
 
 impl RootDatabase {
-    pub fn request_cancellation(&mut self) {
-        let _p = tracing::info_span!("RootDatabase::request_cancellation").entered();
-        self.synthetic_write(Durability::LOW);
-    }
-
     pub fn apply_change(&mut self, change: ChangeWithProcMacros) {
         let _p = tracing::info_span!("RootDatabase::apply_change").entered();
-        self.request_cancellation();
```

Comment:
Notable change here, for one this does not bump the revision anymore (we don't need to, it will be bumped when an actual input gets changed), it just cancels workers. More importantly though, this no longer runs lru eviction which may block.